### PR TITLE
Update dependencies and fix bq tests

### PR DESCRIPTION
**Pull request by keasey-recurve**
### Your checklist for this pull request

Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a feature/bugfix branch** (right side). Don't request your master!
- [x] Make sure tests pass and coverage has not fallen `./test.sh`.
- [x] Update the [CHANGELOG.md](../CHANGELOG.md) to describe your changes in a bulleted list under the "Development" section at the top of the changelog. If this section does not exist, create it.
- [x] Make sure code style follows PEP 008 using `black`.
- [x] Make sure that new functions and classes have inline docstrings.

### Description

\<description of pull request (suggestion: start by copying from your changelog update and elaborate if necessary.)>

### CODE REVIEW

**Comment by keasey-recurve** [comment]
@mattwyatt-recurve I'll grab some time to review this with you next week.

File: setup.py [hunk 1]
```diff
@@ -33,21 +33,12 @@
...
     "sqlalchemy==2.0",
-    "nbformat==5.1.3",
-    "nbconvert==6.3.0",
     "jinja2==3.0.3",
-    "matplotlib<3.6",
-    "numpy<1.23.0",
     "psycopg[binary]==3.1.8",
-    "google-cloud-bigquery==3.6.0",
-    "db-dtypes>=1.0.5"
+    "google-cloud-bigquery>=2.34.3",
```
Comment (mattwyatt-recurve) [inline setup.py:41]: I'm a little worried about this one still, as we know that 3.x.y is out there. But let's cross that bridge when we come to it.

File: tests/test_bq.py [hunk 2]
```diff
@@ -538,10 +560,17 @@ def test_real_data_calculations_time_series(real_data_calculations_time_series):
         table_name="oeem-mcemktpl-platform.cmkt_2023_04_01_flexvalue_metered_outputs.flexvalue_outputs_deer_p2021_ts_elec_latest",
         like_clause=like_clause,
     )
-    prod_df = real_data_calculations_time_series.db_manager._select_as_df(sql=query)
+    result = real_data_calculations_time_series.db_manager._exec_select_sql(sql=query)
```
Comment (mattwyatt-recurve) [inline tests/test_bq.py:563]: This is a good change. Can we take _select_as_df out of db.py?

**Review by mattwyatt-recurve — COMMENTED** [review]
If we can delete `_select_as_df` from db.py, I think we're golden on this.

File: setup.py (hunk 1 above)
Comment (keasey-recurve) [reply setup.py:41]: Sounds good

**Pull request merged**